lib/webdrivers/common.rb

```diff
@@ -9,18 +9,24 @@ def update
         unless site_available?
           return current.nil? ? nil : binary
         end
-        released = latest()
-        location = binary()
+        released      = latest()
+        location      = binary()
+        binary_exists = File.exists?(location)
 
-        return location if released.nil? && File.exist?(location)
+        return location if released.nil? && binary_exists
 
         if released.nil?
           msg = "Unable to find the latest version of #{file_name}; try downloading manually from #{base_url} and place in #{install_dir}"
           raise StandardError, msg
         end
 
-        return location if current == released
-        remove && download
+        if current == released && binary_exists # Already have latest/matching one
```

Comment:
When would `binary_exist` be false? If it doesn't exist, wouldn't `current` throw an exception?

Reply:
Unlike other classes `current` is getting the browser version (Edge) from the Windows registry instead of the binary version. This means that `current` will still return a version when `binary_exists` is false. This is to work around the limitation of `MSWebdriver.exe` not having a `version` argument to retrieve the webdriver version. 

The logic here is to retrieve the Edge version through `current` and then download the corresponding binary if a binary file (any version) does not exist.